sonar-scanner-protocol/src/main/protobuf/scanner_report.proto

```diff
@@ -35,82 +35,117 @@ option java_package = "org.sonar.scanner.protocol.output";
 option optimize_for = SPEED;
 
 message Metadata {
-  optional int64 analysis_date = 1;
+  int64 analysis_date = 1;
   // TODO should we keep this project_key here or not ? Because it's a duplication of Component.key
-  optional string project_key = 2;
-  optional string branch = 3;
-  optional int32 root_component_ref = 4;
-  optional bool cross_project_duplication_activated = 5;
+  string project_key = 2;
+  string branch = 3;
+  int32 root_component_ref = 4;
+  bool cross_project_duplication_activated = 5;
 }
 
 message ActiveRule {
-  optional string rule_repository = 1;
-  optional string rule_key = 2;
-  optional Severity severity = 3;
-  repeated ActiveRuleParam param = 4;
-
-  // TODO replace by map
-  message ActiveRuleParam {
-    optional string key = 1;
-    optional string value = 2;
-  }
+  string rule_repository = 1;
+  string rule_key = 2;
+  Severity severity = 3;
+  map<string,string> params_by_key = 4;
 }
 
 message ComponentLink {
-  optional ComponentLinkType type = 1;
-  optional string href = 2;
+  ComponentLinkType type = 1;
+  string href = 2;
+  
+  enum ComponentLinkType {
+    UNKNOWN_TYPE = 0;
```

Comment:
The suffix `_TYPE` can be removed. Moreover I suggest to rename `UNKNOWN` by `UNSET` everywhere. The term is more exact.